src/openforms/contrib/haal_centraal/clients/brp.py

```diff
@@ -41,13 +41,32 @@ class Person:
 
 
 class BRPClient(PreRequestMixin, ABC, APIClient):
+    def __init__(
+        self,
+        *args,
+        origin_oin: str = "",
+        doelbinding: str = "",
+        verwerking: str = "",
+        gebruiker: str = "",
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        if origin_oin:
+            self.headers["x-origin-oin"] = origin_oin
+        if doelbinding:
+            self.headers["x-doelbinding"] = doelbinding
+        if verwerking:
+            self.headers["x-verwerking"] = verwerking
+        if gebruiker:
+            self.headers["x-gebruiker"] = gebruiker
```

Comment:
I don't know how the requested APIs will behave if they receive a header (even if empty) they don't know. I think it doesn't hurt to have these checks, doesn't cost anything on our side

Reply:
As far as I understood, currently nothing actually works in a real setup because of these missing headers. Can you check what the behaviour is of the test API you have available? That is the only representative environment we currently have.

Standard behaviour is that unknown headers are ignored, reverse proxies add a bunch of headers too, WAF-etc. add headers too, so we can assume that an implementation that doesn't know these headers will just ignore them.